bundle-workflow/python/git/git_repository.py

```diff
@@ -1,30 +1,39 @@
 # Copyright OpenSearch Contributors.
 # SPDX-License-Identifier: Apache-2.0
 
+import os
 import subprocess
 import tempfile
-from dataclasses import dataclass
+import shutil
 
-'''
-This class checks out a Git repository at a particular ref into a temporary directory.
-Clients can obtain the actual commit ID by querying the "sha" attribute, and the temp directory name with "dir.name".
-'''
-@dataclass
 class GitRepository:
-    url: str
-    ref: str
-    sha: str = None
-    dir: tempfile.TemporaryDirectory = None
+    '''
+    This class checks out a Git repository at a particular ref into a named directory (or temporary a directory if no named directory is given). Temporary directories will be automatically deleted when the GitRepository object goes out of scope; named directories will be deleted and recreated when the GitRepository is constructed.
+    Clients can obtain the actual commit ID by querying the "sha" attribute, and the temp directory name with "dir".
+    '''
+    def __init__(self, url, ref, directory = None):
```

Comment:
nit - doesn't the doc need to live after the method declaration?

Reply:
That documentation is for the class